archinstall/lib/disk/device_handler.py

```diff
@@ -0,0 +1,584 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+import time
+from pathlib import Path
+from typing import List, Dict, Any, Optional, TYPE_CHECKING
+
+import parted  # type: ignore
+from parted import Disk, Geometry, FileSystem, PartitionException
+
+from .device_model import DeviceModification, PartitionModification, \
+	BDevice, _DeviceInfo, _PartitionInfo, FilesystemType, Unit, PartitionTable, \
+	ModificationStatus, get_lsblk_info, LsblkInfo, _BtrfsSubvolumeInfo, get_all_lsblk_info, DiskEncryption
+from ..exceptions import DiskError, UnknownFilesystemFormat
+from ..general import SysCommand, SysCallError, JSON
+from ..luks import Luks2
+from ..output import log
+from ..utils.util import is_subpath
+
+if TYPE_CHECKING:
+	_: Any
+
+
+class DeviceHandler(object):
+	_TMP_BTRFS_MOUNT = Path('/mnt/arch_btrfs')
+
+	def __init__(self):
+		self._devices: Dict[Path, BDevice] = {}
+		self.load_devices()
+
+	@property
+	def devices(self) -> List[BDevice]:
+		return list(self._devices.values())
+
+	def load_devices(self):
+		block_devices = {}
+
+		for device in parted.getAllDevices():
+			disk = Disk(device)
```

Comment:
```suggestion
			try:
				disk = Disk(device)
			except DiskLabelException as error:
				if 'unrecognised disk label' in getattr(error, 'message', str(error)):
					disk = freshDisk(device, 'gpt')
				else:
					continue
```

This is a possible way to fix https://github.com/archlinux/archinstall/pull/1604#issuecomment-1509924019

If you use this don't forget the imports:

```diff
-from parted import Disk, Geometry, FileSystem, PartitionException
+from parted import Disk, freshDisk, Geometry, FileSystem, PartitionException
+from _ped import DiskLabelException
```

Reply:
I managed to reproduce the error and this seems to fix it indeed! 
I guess if the label is not recognizable it is save to fallback to a fresh disk :thinking:

Reply:
I think a device not having a label means that it does not have a partition table and/or is not formatted but I did not look into it. Using `parted.freshDisk()` does not appear to touch the disk so the fallback should be safe.